### PR TITLE
WIP - feature: allow to share wifi access via QR (2-Step download)

### PR DESCRIPTION
**Pull request by andi34**
Changes by Andreas Blaesius

- api(qr): only pass qr config
- modal(qr): make translatable
- modal(qr): allow using custom text
- qr: fix on gallery and livechroma
- lang: add missing manual entries for WiFi QR
- linting

October 2023:
- adjust code to match latest code style

Change-Id: I12f2d682cf0c1c9d8d3aa1d16b000b6e38016c5a

### CODE REVIEW

File: api/qr.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<?php
```
Comment (benjaminkott) [inline api/qr.php:1]: Do not move classes into the API structure, keep it simple as control structures that call classes
for example, move everything to QrCodeUtility and make function static.

File: api/qr.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            $sec = (string) 'T:WPA;';
+        } elseif ($qrConfig['wifi_secure'] == 'WEP') {
+            $sec = (string) 'T:WEP;';
+        } else {
+            $sec = (string) '';
+        }
+        $wifi = 'WIFI:S:' . $qrConfig['wifi_ssid'] . ';' . $sec . 'P:' . $qrConfig['wifi_pass'] . $hidden . ';';
+
+        $result = QrCodeUtility::create($wifi);
+        header('Content-Type: ' . $result->getMimeType());
```
Comment (benjaminkott) [inline api/qr.php:27]: when moved to QrCodeUtility only return the QRCode, keep responses for the API endpoints—that way the codes can be utilized and generated for multiple purposes.

File: .gitignore [hunk 3]
```diff
@@ -1,4 +1,5 @@
 archives/
+api/QR_ECLEVEL*
```
Comment (benjaminkott) [inline .gitignore:2]: not sure what that file is for

File: template/components/modal.qr.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
+<div id="newQrModal" class="w-full h-full fixed top-0 left-0 hidden place-items-center p-4 [&.isOpen]:grid" style="z-index:9999">
+    <div class="w-full h-full absolute top-0 left-0 z-10 bg-black/60 cursor-pointer" onclick="closeQrCodeModal();"></div>
```
Comment (benjaminkott) [inline template/components/modal.qr.php:8]: tailwind has just been remove from the frontend, and should not be introduced again
use current modal classes.

File: .gitignore (hunk 3 above)
Comment (andi34) [reply .gitignore:2]: Leftover

File: api/qr.php (hunk 1 above)
Comment (andi34) [reply api/qr.php:1]: Just started to adjust the existing commit to apply/work. There's a lot more to do 🙈

**Pull request closed without merging**